module/VuFind/src/VuFind/Db/Service/UserResourceService.php

```diff
@@ -127,4 +138,107 @@ public function getStatistics(): array
         $stats = current($query->getResult());
         return $stats;
     }
+
+    /**
+     * Unlink rows for the specified resource. This will also automatically remove
+     * any tags associated with the relationship.
+     *
+     * @param User|int          $user        ID of user removing links
+     * @param string|array|null $resource_id ID (or array of IDs) of resource(s) to
+     *                                       unlink (null for ALL matching
+     *                                       resources)
+     * @param UserList|null     $list        list to unlink (null for ALL matching lists, with the destruction
+     *                                       of all tags associated with the $resource_id value; true for ALL
+     *                                       matching lists, but retaining any tags associated with the
+     *                                       $resource_id independently of lists)
+     *
+     * @return void
+     */
+    public function destroyLinks($user, $resource_id, $list = null)
+    {
+        // Remove any tags associated with the links we are removing; we don't
+        // want to leave orphaned tags in the resource_tags table after we have
+        // cleared out favorites in user_resource!
+        $tagService = $this->getDbService(\VuFind\Db\Service\TagService::class);
+        $tagService->destroyResourceLinks($resource_id, $user, $list);
+
+        $dql = 'DELETE FROM ' . $this->getEntityClass(UserResource::class) . ' ur ';
+        $dqlWhere = ['ur.user = :user '];
+        $parameters = compact('user');
+        if (null !== $resource_id) {
+            if (!is_array($resource_id)) {
```

Comment:
Instead of this if statement, you can just cast to an array in the assignment below:

```
$parameters['resource_id'] = (array)$resource_id;
```